guides/source/active_record_migrations.md

```diff
@@ -647,31 +797,55 @@ end
 ```
 
 INFO: Tables with composite primary keys require passing array values rather
-than integer IDs to many methods. See also the [Active Record Querying](
-active_record_querying.html) guide to learn more.
+than integer IDs to many methods. See also the [Active Record Composite Primary
+Keys](active_record_composite_primary_keys.html) guide to learn more.
 
-### When Helpers aren't Enough
+### Execute Arbitrary SQL
 
-If the helpers provided by Active Record aren't enough you can use the [`execute`][]
-method to execute arbitrary SQL:
+If the helpers provided by Active Record aren't enough, you can use the
+[`execute`][] method to execute arbitrary SQL. For example,
 
 ```ruby
-Product.connection.execute("UPDATE products SET price = 'free' WHERE 1=1")
+class UpdateProductPrices < ActiveRecord::Migration[8.0]
+  def up
+    execute "UPDATE products SET price = 'free' WHERE 1=1;"
+  end
+
+  def down
+    execute "UPDATE products SET price = 'original_price' WHERE price = 'free';"
+  end
+end
 ```
 
-For more details and examples of individual methods, check the API documentation.
+In this example, we're updating the `price` column of the products table to
+'free' for all records.
+
+WARNING: Modifying data directly in migrations should be approached with
+caution. Consider if this is the best approach for your use case, and be aware
+of potential drawbacks such as increase complexity and maintenance overhead,
```

Comment:
I think this is meant to say "increased" (past tense). 

However, I wonder if this section could be more valuable by being more specific. For example, data modification statements in migrations aren't aware of table row counts in prod. Something fast in dev could be slow in prod.

For large tables, long running statements might be worth performing outside of Rails migrations, so that they can be performed during a low activity period, possibly as batched operations.

Reply:
I'm not sure if data migrations is in-scope for this guide (could be a separate guide).
There also isn't a recommended Rails-way for data migrations, that I'm aware of.